### PR TITLE
add funder

### PR DESCRIPTION
**Pull request by anhphi257**

### CODE REVIEW

File: programs/staking/src/lib.rs [hunk 1]
```diff
@@ -189,6 +189,27 @@ mod staking {
 
         Ok(())
     }
+
+    pub fn change_funder(ctx: Context<FunderChange>, funder: Pubkey) -> Result<()> {
+        let vault = &mut ctx.accounts.vault;
+        if funder == vault.funder.key() {
```
Comment (andrewsource147) [inline programs/staking/src/lib.rs:195]: We can add a macro check for this in the context

File: programs/staking/src/lib.rs [hunk 2]
```diff
@@ -189,6 +189,27 @@ mod staking {
...
+            return Err(VaultError::FunderAlreadyAuthorized.into());
+        }
+        vault.funder = funder;
+        Ok(())
+    }
+
+    pub fn deauthorized_funder(ctx: Context<FunderChange>, funder: Pubkey) -> Result<()> {
```
Comment (andrewsource147) [inline programs/staking/src/lib.rs:202]: I think we dont need this function, If we want to deauthorized_funder, we just change_funder to Pubkey::default()

File: programs/staking/src/lib.rs [hunk 3]
```diff
@@ -300,6 +324,16 @@ pub struct TransferAdmin<'info> {
     pub new_admin: AccountInfo<'info>,
 }
 
+#[derive(Accounts)]
+pub struct FunderChange<'info> {
+    #[account(
+    mut,
+    has_one = admin,
```
Comment (andrewsource147) [inline programs/staking/src/lib.rs:331]: pls check to prettify the format

Comment (anhphi257) [reply programs/staking/src/lib.rs]: resolved

File: programs/staking/src/lib.rs (hunk 2 above)
Comment (anhphi257) [reply programs/staking/src/lib.rs]: resolved

**Pull request merged**